diff --git a/src/frontend/mame/ui/ui.cpp b/src/frontend/mame/ui/ui.cpp
index fd59bb3d..8697b359 100644
--- a/src/frontend/mame/ui/ui.cpp
+++ b/src/frontend/mame/ui/ui.cpp
@@ -815,7 +815,7 @@ void mame_ui_manager::display_startup_screens(bool first_time)
 					if (need_warning)
 						m_last_warning_time = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
 				}
-				if (need_warning)
+				if (need_warning && !options().skip_warnings())
 				{
 					warning_text = machine_info().warnings_string();
 					warning_text.append(_("\n\nPress any key to continue"));
